testproject/Assets/Tests/Manual/Scripts/ConnectionApprovalComponent.cs

```diff
@@ -165,14 +165,17 @@ private void ConnectionApprovalCallback(byte[] dataToken, ulong clientId, Networ
 
             if (m_GlobalObjectIdHashOverride != 0 && m_PlayerPrefabOverride && m_PlayerPrefabOverride.isOn)
             {
-                aprovalCallback.Invoke(true, m_GlobalObjectIdHashOverride, isTokenValid, null, null);
+                decision.Approved = isTokenValid;
```

Comment:
Those three lines should be the first focus. The API changed in this way: instead of asking the client code to call a callback, they fill a class with their decision. All the approval handler get to run before the decision is used. Then, the decision is used just once.